### PR TITLE
Turn on TreatWarningsAsErrors since they are currently harmless

### LINKED ISSUE #24755: Quarantined test: LinkedApplicationTests.LinkedApplicationWorks
Build failure: https://dev.azure.com/dnceng/public/_build/results?buildId=763379&view=ms.vss-test-web.build-test-results-tab&runId=23716264&resultId=100000&paneView=debug

```C#
 at System.Net.Http.HttpConnection.ParseStatusLine(Span`1 line, HttpResponseMessage response)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.IntegrationTesting.LoggingHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /_/src/Hosting/Server.IntegrationTesting/src/Common/LoggingHandler.cs:line 23
   at System.Net.Http.HttpClient.FinishSendAsync(ValueTask`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts, Boolean buffered, Boolean async, CancellationToken callerToken, Int64 timeoutTime)
   at Microsoft.AspNetCore.Server.IntegrationTesting.RetryHelper.RetryRequest(Func`1 retryBlock, ILogger logger, CancellationToken cancellationToken, Int32 retryCount) in /_/src/Hosting/Server.IntegrationTesting/src/Common/RetryHelper.cs:line 39
   at Microsoft.AspNetCore.Server.IntegrationTesting.RetryHelper.RetryRequest(Func`1 retryBlock, ILogger logger, CancellationToken cancellationToken, Int32 retryCount) in /_/src/Hosting/Server.IntegrationTesting/src/Common/RetryHelper.cs:line 55
   at Microsoft.AspNetCore.Hosting.FunctionalTests.LinkedApplicationTests.LinkedApplicationWorks() in /_/src/Hosting/test/FunctionalTests/LinkedApplicationTests.cs:line 51
--- End of stack trace from previous location ---
```

### PR DESCRIPTION
**Pull request by davidfowl**
Attempting to fix https://github.com/dotnet/aspnetcore/issues/24755

### CODE REVIEW

**Comment by Tratcher** [comment]
Was anything done to fix LinkedApplicationWorks?

**Comment by davidfowl** [comment]
No I'm just turning it back on to see if it resolved itself. Lots of changes were made to .NET Core wrt trimming. Though I might take a look at how aggressively this trims and scale it back a bit.

Here are the list of warnings that need to be fixed (well some of them need to be suppressed):
https://gist.github.com/davidfowl/ceaecbfa487d4bd694671ce3b38c0088

**Review by pranavkm — APPROVED** [review]

**Comment by eerhardt** [comment]
> Here are the list of warnings that need to be fixed (well some of them need to be suppressed)

Do we have an issue to track resolving these?

File: src/Hosting/test/testassets/BasicLinkedApp/BasicLinkedApp.csproj [hunk 1]
```diff
@@ -5,6 +5,7 @@
     <OutputType>Exe</OutputType>
     <PublishTrimmed>true</PublishTrimmed>
     <TrimMode>link</TrimMode>
+    <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
```
Comment (eerhardt) [inline src/Hosting/test/testassets/BasicLinkedApp/BasicLinkedApp.csproj:8]: How about just `<SuppressTrimAnalysisWarnings>true</SuppressTrimAnalysisWarnings>`? So other warnings are still errors.

Comment (davidfowl) [reply src/Hosting/test/testassets/BasicLinkedApp/BasicLinkedApp.csproj:8]: I'll do that, but I do want to file issues for the warnings so I'd like them to show up.

Comment (eerhardt) [reply src/Hosting/test/testassets/BasicLinkedApp/BasicLinkedApp.csproj:8]: How about `<ILLinkTreatWarningsAsErrors>false</ILLinkTreatWarningsAsErrors>`

See https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming-options#suppressing-warnings

**Review by eerhardt — APPROVED** [review]

**Pull request merged**